### DISCUSSION TITLE
Harvesting addresses on every emacs restart takes a very long time

### DISCUSSION THREAD

**Discussion by newhallroad** [tarsius/notmuch-addr#3] [Support] [answered]
The first time notmuch-addr tries to autcomplete an email address after restarting emacs, the "Collecting email addresses..." process takes a very long time, apparently running from scratch. During this process, emacs hangs. Once it is complete, things run very smoothly and quickly during subsequent autocompletes, once `notmuch-addr--cache` has been set. The initial delay, however, is so long that it makes the package almost unusable. 

I understand the need for the wait one initial time, as the script has run through my 30GB of emails. But is there a way to avoid the wait every time emacs starts? Perhaps some kind of caching? Or making use of saving the addresses at `notmuch-address-save-filename` or somewhere similar? 

I would be happy to help you try out alternatives if that would be useful. Or, if I am missing something (more than possible!) please let me know.

With great thanks and appreciation.

**Comment by tarsius** [comment, marked as answer]
For getting the list of addresses is so faster, using a persistent cache wouldn't make sense:

```
$ time notmuch address --format=text --format-version=4 --output=recipients --deduplicate=address "from:jonas@bernoul.li or from:jonasbernoulli@gmail.com" > /dev/null

real    0m0.378s
user    0m0.301s
sys     0m0.077s
```

My maildir is only 5GB, but even if we multiply the above by six, that would still be fast enough.

Maybe newer notmuch versions can produce this information quickly, but you are using an older release, which doesn't.
